index.js

```diff
@@ -24,67 +25,84 @@ function getRandomNumber(min, max) {
 
 function checkGuess() {
   // Get value from guess input element
-  const guess = parseInt(guessInput.value, 10);
-  attempts = attempts + 1;
-
+  let guess = parseInt(guessInput.value, 10);
+  console.log(guess)
+  
   hideAllMessages();
 
+
   if (guess === targetNumber) {
-    numberOfGuessesMessage.style.display = '';
+
+    
     numberOfGuessesMessage.innerHTML = `You made ${attempts} guesses`;
+    numberOfGuessesMessage.style.display ='';
 
-    correctMessage.style.display = '';
 
     submitButton.disabled = true;
     guessInput.disabled = true;
   }
 
-  if (guess !== targetNumber) {
-    if (guess < targetNumber) {
-      tooLowMessage.style.display = '';
-    } else {
-      tooLowMessage.style.display = '';
-    }
+  if (guess < targetNumber) {
+    tooLowMessage.style.display = '';
+  }
+  if (guess > targetNumber) {
+    tooHighMessage.style.display = '';
+  }
 
-    const remainingAttempts = maxNumberOfAttempts - attempts;
+  {
 
-    numberOfGuessesMessage.style.display = '';
-    numberOfGuessesMessage.innerHTML = `You guessed ${guess}. <br> ${remainingAttempts} guesses remaining`;
-  }
 
-  if (attempts ==== maxNumberOfAttempts) {
-    submitButton.disabled = true;
-    guessInput.disabled = true;
+
+
+
+    const remainingAttempts = maxNumberOfAttempts - attempts;
+    console.log(remainingAttempts)
   }
 
+
+
+  maxGuessesMessage.innerHTML = `You guessed ${guess} <br>   guesses remaining `;
+  maxGuessesMessage.style.display = '';
+
+}
+
+if (attempts === maxNumberOfAttempts) {
+
+  submitButton.disabled = true;
+  guessInput.disabled = true;
+
+
   guessInput.value = '';
 
   resetButton.style.display = '';
 }
 
 function hideAllMessages() {
-  for (let elementIndex = 0; elementIndex <= messages.length; elementIndex++) {
+  for (let elementIndex = 0; elementIndex < messages.length; elementIndex++) {
     messages[elementIndex].style.display = 'none';
+
   }
+
 }
 
-funtion setup() {
+function setup() {
   // Get random number
   targetNumber = getRandomNumber(1, 100);
-  console.log(`target number: ${targetNumber}`);
+  console.log(`targetNumber`);
 
   // Reset number of attempts
-  maxNumberOfAttempts = 0;
+  console.log(maxNumberOfAttempts)
 
   // Enable the input and submit button
   submitButton.disabeld = false;
   guessInput.disabled = false;
 
   hideAllMessages();
-  resetButton.style.display = 'none';
+  resetButton.style.display = '';
```

Comment:
Its not letting me click on the attach video . its not clickable

Reply:
sorry about that; I think it didn't attach properly. Here's a link: https://www.veed.io/view/bba9ebc4-6fce-48a0-aeb4-92dc81ce11be?panel=share

Reply:
when i click ctrl  s to save the  display:none; in css styles section. it gives me an option to make a new folder. should i replace it with a new folder?

Reply:
I'm not sure what you mean by CSS styles section. You should be able to just update it here in this `index.js` file; so instead of what you have on line `101` (`resetButton.style.display = ''`); you'd want to set `display` to `none`

Reply:
I am having trouble. the remaining guesses reach 0. but if I type in another number the guesses remaining go into negative numbers.  Also, when I guess the correct number. the input field and submit guess button goes disable. however, when I click reset it doesn't go back to its initial state.

Reply:
> I am having trouble. the remaining guesses reach 0. but if I type in another number the guesses remaining go into negative numbers.

Nice, it sounds like you're making good progress on incrementing the `attempts` variable for each guess. So there's there section in the code that should be checking "has the user guessed the max amount of times? If yes, then disable the submit button so that they can't enter anymore guesses": [https://github.com/Code-the-Dream-School/intro-to-programming-section-5/pull/23/files#diff-e727e4bdf3657fd1d798edcd6b099d6e092f8573cba266154583a746bba0f346R69-R78](https://github.com/Code-the-Dream-School/intro-to-programming-section-5/pull/23/files#diff-e727e4bdf3657fd1d798edcd6b099d6e092f8573cba266154583a746bba0f346R69-R78)

Is that part not happening for you? Remember that Javascript reads code from top to bottom, so if other things are happening before it gets to that `if` statement, then maybe you can rearrange the statements. 

> Also, when I guess the correct number. the input field and submit guess button goes disable. however, when I click reset it doesn't go back to its initial state.

Make sure that:

*   Javascript is correctly finding the right element for the `resetButton` element: [https://github.com/Code-the-Dream-School/intro-to-programming-section-5/pull/23/files#diff-e727e4bdf3657fd1d798edcd6b099d6e092f8573cba266154583a746bba0f346R3](https://github.com/Code-the-Dream-School/intro-to-programming-section-5/pull/23/files#diff-e727e4bdf3657fd1d798edcd6b099d6e092f8573cba266154583a746bba0f346R3)
*   Javascript is listening for a `"click"` event so that it knows when a user clicks that element
*   The function that is run on the `"click"` listener is able to successfully find and re-enable all of the expected inputs/buttons

Remember to use `console.log` or other debugging approaches to work through this.

If you commit and push what you have so far, I can take a look at what you currently have and give advice based on that.

Reply:
ok i just commit and push

Reply:
Ah okay I see, so it is correctly disabling the Submit button and the input box when the user correctly guesses.

But when you click reset, it's _**only**_ re-enabling the input box, but not the submit button. 

Can you find where in the code the Submit button is being **disabled**? How about when it's being **enabled** (remember this might be setting `disabled` property to `false`). Are there any bugs in that part of the code? Is everything written and typed correctly?

Reply:
if(guess === targetNumber){
      submitButton.disabled = true;
      guessInput.disabled = true;
      guessInput.value = 'Submit Guess';
      maxNumberOfAttempts.innerHTML="You guessed Correctly"
      
  }

Reply:
this disable submit  and input when i guess correct number

Reply:
if (guess === targetNumber) {

    
    numOfGuessesMessage.innerHTML = `You made ${attempts} guesses`;
  
    correctMessage.style.display ='';


    submitButton.disabled = true;
    guessInput.disabled = true;
    resetButton.style.display=''
    
  }

Reply:
Are these two the same thing?

Reply:
```javascript
if(guess === targetNumber) {
	submitButton.disabled = true;
	guessInput.disabled = true;
}
```

Yep, you're correct! This will disable both the submit button and the input button when you guess the correct number :white\_check\_mark: 

Now, where does the opposite happen? Where do they get **enabled** ?

Reply:
Yes both if statements are basically doing the same thing. They're running when the user guesses the number correctly. One you have earlier in your function which is good.

The one you have lower in the function has a few things that caught my eye:  
 

```
guessInput.value = 'Submit Guess';
maxNumberOfAttempts.innerHTML="You guessed Correctly"
```

`guessInput` is an input box element for **numbers** but you're setting it's value to a **string**. You might have noticed a warning in your developer console about that. 

`maxNumberOfAttempts` is a variable that was storing a **number** but you're setting an `innerHTML` on it as if it was an **HTML Element**. Do we want to ever change `maxNumberOfAttempts` ?